### ISSUE TITLE
Failed to pull image "bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4"

### ISSUE THREAD

**Issue by mobius7777** [Azure/Bridge-To-Kubernetes#634]
I've been struggling for hours to get Bridge To Kubernetes to work in VS2022 for a kubernetes project. It had been working perfectly fine, but earlier today, running the service I was debugging via B2K simply failed to load the service. According to k9s, the service is bouncing between "ErrImagePull" and "ImagePullBackOff". 

Checking  `kubectl describe pod` for the service reveals the following error:

`Failed to pull image "bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4": Error response from daemon: manifest for bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4 not found: manifest unknown: manifest tagged by "1.3.4" is not found`

If I docker pull this myself, I get the same error.
`docker pull bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4`
`Error response from daemon: manifest for bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4 not found: manifest unknown: manifest tagged by "1.3.4" is not found`

It also happens if I try to get by latest tag.

I have no idea what to do.

**Comment by xuexiaoxuan** [comment]
Have you solved your problem? I have the same problem

**Comment by mobius7777** [comment]
@xuexiaoxuan  Sadly, I have not. It continues to fail for me the same way. I am basically stuck. I have no idea how or why this suddenly started, but the problem started late Friday afternoon. 

I guess it's good to know I'm not the only one having the issue. We can commiserate together...

Maybe someone will have an answer on Monday.

**Comment by mgszy** [comment]
Just tried to use b2k this morning and I am also getting this issue :(

**Comment by ace90210** [comment]
I am getting the same issue if anyone finds a fix please share it

**Comment by davesmith999** [comment]
I'm having the same issue

**Comment by mobius7777** [comment]
Why would they remove the old ARC before updating and releasing a new version? I can't make B2K pull the images from somewhere else, can I? I don't even know where the new ACR is.

**Comment by pkoelbl** [comment]
You can point it at other ACRs using these 2 environment variables
`BRIDGE_DEVHOSTIMAGENAME`
`BRIDGE_RESTORATIONJOBIMAGENAME`

However it appears we can't point it at the new repo yet.

```
Error response from daemon: Head "https://bridgetok8s.azurecr.io/v2/lpkremoteagent/manifests/1.3.4": unauthorized: {"errors":[{"code":"UNAUTHORIZED","message":"authentication required, visit https://aka.ms/acr/authorization for more information."}]}
```

**Comment by mobius7777** [comment]
So we are just out of luck and dead in the water for now?

**Comment by stevendarby** [comment]
@pkoelbl going by the PR, the authentication method changed too

**Comment by stevendarby** [comment]
Issue is affecting me too... any change of looking at this @hsubramanianaks?

**Comment by hsubramanianaks** [comment]
Acknowledged, I will look into this and get back soon. Thanks for your patience.

**Comment by sivasankaritcs** [comment]
Hi All,
I am also getting the same,If anyone solved, please do describe here. 

 "Failed to pull image "bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4": Error response from daemon: manifest for bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4 not found: manifest unknown: manifest tagged by "1.3.4" is not found"

**Comment by steverash** [comment]
Adding the 3 env variables pointing to the new ACR seems to work (it downloads the images now), but I do notice there seems to be a problem with the actual images

```
You must install or update .NET to run this application.

App: /app/restorationjob
Architecture: x64
Framework: 'Microsoft.NETCore.App', version '7.0.15' (x64)
.NET location: /usr/share/dotnet

The following frameworks were found:
  8.0.1 at [/usr/share/dotnet/shared/Microsoft.NETCore.App]

Learn more:
https://aka.ms/dotnet/app-launch-failed

To install missing framework, download:
https://aka.ms/dotnet-core-applaunch?framework=Microsoft.NETCore.App&framework_version=7.0.15&arch=x64&rid=linux-x64&os=mariner.2.0
```


fyi - these are the 3 I added (the image names are guesswork, so please correct if I'm wrong)

BRIDGE_DEVHOSTIMAGENAME = bridgetok8s.azurecr.io/lpkremoteagent:1.3.4
BRIDGE_RESTORATIONJOBIMAGENAME = bridgetok8s.azurecr.io/lpkrestorationjob:1.3.4
BRIDGE_ROUTINGMANAGERIMAGENAME = bridgetok8s.azurecr.io/routingmanager:stable

**Comment by nam-osisoft** [comment]
those above env vars (BRIDGE_DEVHOSTIMAGENAME BRIDGE_RESTORATIONJOBIMAGENAME  BRIDGE_ROUTINGMANAGERIMAGENAME) did NOT work for me... got the same stuck connecting error message.

**Comment by letmagnau** [comment]
mee too... no  fix... how can I set this env vars??

**Comment by steverash** [comment]
I had to restart VS before they took effect.  In fact to be safe the first time I rebooted the PC (VS seemed to be a bit stuck).  

I just set the env vars in the system environment variables control panel (https://learn.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_environment_variables?view=powershell-7.4#set-environment-variables-in-the-system-control-panel).

**Comment by mobius7777** [comment]
Can you just put the images back on bridgetokubernetes.azurecr.io until a new version of the VS is released that points to the new ARC?

**Comment by mobius7777** [comment]
@hsubramanianaks I got the same error as @stevendarby . Can we please have the images back on bridgetokubernetes.azurecr.io as they were before so we can continue our jobs?

**Comment by hsubramanianaks** [comment]
The old ACR repositories are back, can you all please retest and confirm here?

**Comment by martinmiranda14** [comment]
I have the same problem with the bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.2 image

**Comment by sivasankaritcs** [comment]
Hi Everyone,
Step 1:
$env:BRIDGE_DEVHOSTIMAGENAME="bridgetok8s.azurecr.io/lpkremoteagent:1.3.4"
$env:BRIDGE_RESTORATIONJOBIMAGENAME="bridgetok8s.azurecr.io/lpkrestorationjob:1.3.4"
$env:BRIDGE_ROUTINGMANAGERIMAGENAME="bridgetok8s.azurecr.io/routingmanager:stable"

I have created Environmetal Variables and I have pulled all the 3 images

Now I can debug using BridgetoK8S.
Its perfectly works for me.
Thanks

**Comment by sivasankaritcs** [comment]
Yes,it is back "bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4"
docker pull bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4
1.3.4: Pulling from lpkremoteagent
Digest: sha256:82c0039c90d4cdc6bdb8614d58be2e9ddb4a06ec8156c33031e5bb87dbec28bd
Status: Image is up to date for bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4
bridgetokubernetes.azurecr.io/lpkremoteagent:1.3.4
![image](https://github.com/Azure/Bridge-To-Kubernetes/assets/22616177/2b7bfe76-99cc-4ef4-b4ac-1d10e56bae75)

**Comment by mobius7777** [comment]
@hsubramanianaks I can debug again, thank you!
How is this going to work in the future? Obvious you are wanting to migrate to bridgetok8s.azurecr.io, so what can we expect? Will the images there be fixed, since they are giving the "You must install or update .NET to run this application" error?

**Comment by hsubramanianaks** [comment]
@mobius7777 Thanks for testing it. this switch was not intended to be turned live yet, but some internal security issues caused the repos to be deleted which is reverted now. When we switch to new acr repo, users will not see any issues like this is our best hope. We will keep you posted on this. For now can we close this issue?

**Comment by nam-osisoft** [comment]
works now.. i did not even have to set env vars. Thanks

**Comment by mobius7777** [comment]
@hsubramanianaks I am not having any more issues since you reactivated the bridgetokubernetes.azurecr.io ARC. Thank you. I guess you can close the issue since that resolved it.

**Comment by hsubramanianaks** [comment]
Closing this issue as it works for everyone. Please feel free to reopen if anyone face issues. Thank you all.

**Comment by krosn** [comment]
I am seeing this same issue again as of about half an hour ago

Back-off pulling image "bridgetokubernetes.azurecr.io/lpkrestorationjob:1.3.4"

**Comment by stevendarby** [comment]
@krosn this project is dead, see the stickied posts

**Comment by krosn** [comment]
Thanks
